modules/security/keyvault/main.bicep

```diff
@@ -0,0 +1,100 @@
+@description('Deployment Location')
+param location string
+
+@description('Prefix of Cosmos DB Resource Name')
+param prefix string = 'kv'
+
+@description('Name of the Key Vault')
+param name string = '${prefix}-${uniqueString(resourceGroup().id)}'
+
+@description('The tenant ID where the Key Vault is deployed')
+param tenantId string = subscription().tenantId
+
+@description('For an existing Managed Identity, the Subscription Id it is located in')
+param subscriptionId string = subscription().subscriptionId
+
+@description('For an existing Managed Identity, the Resource Group it is located in')
+param resourceGroupName string = resourceGroup().name
+
+@description('Subnet ID for the Key Vault')
+param subnetID string = ''
+
+@description('List of RBAC policies to assign to the Key Vault')
+param rbacPolicies array = []
+
+@description('RBAC Role Assignments to apply to each RBAC policy')
+param roleAssignments array = [
+  '4633458b-17de-408a-b874-0445c86b69e6' // rbacSecretsReaderRole
+  'a4417e6f-fecd-4de8-b567-7b0420556985' // rbacCertificateOfficerRole
+]
+
+@allowed([ 'new', 'existing' ])
+@description('Whether to create a new Key Vault or use an existing one')
+param newOrExisting string = 'new'
+
+@description('Name of Secret to add to Key Vault')
+param secretName string = 'secret${uniqueString(resourceGroup().id)}'
+
+@secure()
+@description('Value of Secret to add to Key Vault')
+param secretValue string = ''
+
+@description('Specifies whether soft delete should be enabled for the Key Vault.')
+param enableSoftDelete bool = true
+
+@description('The number of days to retain deleted data in the Key Vault.')
+param softDeleteRetentionInDays int = 7
+
+@allowed(['standard', 'premium'])
+@description('The SKU name of the Key Vault.')
+param skuName string = 'standard'
+
+@allowed(['A', 'B'])
+@description('The SKU family of the Key Vault.')
+param skuFamily string = 'A'
+
+@description('Specifies whether RBAC authorization should be enabled for the Key Vault.')
+param enableRbacAuthorization bool = true
+
+module keyVault 'modules/vaults.bicep' = {
+  name: guid(name, 'deploy')
+  scope: resourceGroup(subscriptionId, resourceGroupName)
```

Comment:
Including this in Bicep can cause conflicts based on how the module is consumed. I do not see other modules using this and being that a KV only resides in a given Resource Group I'm not sure there is a need to have scope defined.

Reply:
I was pushing this module a little further than the rest and was aiming to make it as generic and universal as possible. This is the first module I have gone this far with, so good spot here for some extra feedback and review.

My goal here was that this module should be able to create a new key vault in any sub/rg, or use an existing one, and then add a secret to it.

Without this part, I can add a secret to an existing key vault in a different RG. But if the user provides me with a different sub/resource group, and also uses the "new" flag, even if they have permission to perform the action, it will still fail.

Reply:
@wsmelton , do you have an example in mind that would cause this to fail? I can try and generate a test case for us that covers this so we can make sure the module addresses the concern.

Reply:
To deploy Bicep you have to connect to a specific subscription. Why do you need to support cross deployments?

Anyone using this to create or update secrets will be in the same subscription and resource group of the Key Vault. There is no reason for them not to be the way the deployment command works.

Reply:
The general pattern promoted by most is Key Vault for each application or at most Resource Group. I'm not sure anyone uses a central or shared KV because you only control access at the vault level.

Reply:
rolled back the rg/sub parameters and will think about this again when I have the use case in front of me.